app/src/main/java/com/example/weatherapp/viewmodel/WeatherDetailsViewModel.kt

```diff
@@ -12,21 +14,37 @@ import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.withContext
 
-class WeatherDetailsViewModel: ViewModel() {
+class WeatherDetailsViewModel(
+    private val savedStateHandle: SavedStateHandle,
+    private val city: String,
+): ViewModel() {
     private val _uiState = MutableStateFlow(WeatherDetailsUiState())
     val uiState: StateFlow<WeatherDetailsUiState> = _uiState.asStateFlow()
     private val locationWeatherRepository = WeatherDetailsRepository()
+    private val savedStateHandle_LOCATION_KEY = "location"
 
     fun isPageLoaded(loaded: Boolean ){
         _uiState.update { currentState ->
             currentState.copy(isPageLoaded = loaded)
         }
     }
 
-    suspend fun setTemperatureDetails(location: String) {
+    init {
+        setLocation(city)
+    }
+
+    private fun setLocation(location: String){
+        savedStateHandle[savedStateHandle_LOCATION_KEY] = location
```

Comment:
Why do we need to save the `location` in the `savedStateHandle`? Or, more general, what is the `SavedStateHandle` purpose in this ViewModel?

Reply:
Location is no longer used in this way now🕺
I have passed in an argument with the key "Location"
My savedStateHandle has this value from the moment my viewModel is active. Therefore I no longer have a setLocation function